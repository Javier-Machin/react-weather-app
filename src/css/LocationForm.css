.location-form {
  padding: 4rem 3rem;
  background-color: rgba(50, 160, 180, 0.7);
  border-radius: 0 0 4px 4px;
  display: flex;
  align-items: center;
  border: 1.8px solid rgba(0,0,0,.78);
  border-width: 0 1.8px 1.8px 1.8px;
  justify-content: space-around;
}

.location-form__input {
  font-size: 2rem;
  border: 0;
  text-decoration: none;
  outline: none;
  padding: 1rem 4rem;
  width: 30%;
  border-radius: 5px;
  border-bottom: 3px solid transparent;
  color: rgba(0,0,0,.8);
  letter-spacing: .2rem;
  font-weight: 500;
  text-transform: uppercase;
  transition: all .5s;
}

.location-form__input:focus {
  border-bottom: 3px solid rgba(245,96,44,1);
  border-radius: 5px 5px 0 0;
}

.location-form__input::-webkit-input-placeholder {
  color: rgba(0,0,0,.4);
  letter-spacing: .2rem;
  font-weight: 500;
  text-transform: uppercase;
}

.location-form__btn {
  background-color: rgba(25,170,25,.9);
  font-size: 2rem;
  border: 0;
  cursor: pointer;
  padding: 1rem 2rem;
  border-radius: 15px;
  color: white;
  letter-spacing: .2rem;
  text-transform: uppercase;
  font-weight: 500;
  transition: all .4s;
}

.location-form__btn:hover {
  transform: translateY(-2px);
  box-shadow: 0 .7rem 1rem rgba(0,0,0,.2); 
}

.location-form__btn:active {
  transform: translateY(2px);
  box-shadow: 0 .7rem 1rem rgba(0,0,0,.3); 
}

.location-form__btn:focus {
  outline:0;
  box-shadow: 0 0 3px 2px rgba(245,96,44,1); 
}