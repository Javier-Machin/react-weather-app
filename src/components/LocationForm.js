import React, { useState } from 'react';
import '../css/LocationForm.css';

// Form to enter the location for the weather request

const LocationForm = props => {
  // Use state hook
  const [location, setLocation] = useState('');

  const handleChange = event => {
    setLocation(event.target.value);
  };

  const handleSubmit = event => {
    event.preventDefault();
    props.handleLocationChange(location);
    // Reset input after submit
    event.target.reset();
  };

  return (
    <form className="location-form" onSubmit={handleSubmit}>
      <input
        placeholder="enter a city"
        className="location-form__input"
        name="location"
        onChange={handleChange}
        type="text"
      />
      <input className="location-form__btn" type="submit" value="Check weather &rarr;" />
    </form>
  );
};

export default LocationForm;
