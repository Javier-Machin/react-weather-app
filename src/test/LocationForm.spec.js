import React from 'react';
import { shallow } from 'enzyme';
import LocationForm from '../components/LocationForm';

describe('LocationForm', () => {
  it('should render correctly', () => {
    const component = shallow(<LocationForm />);
    expect(component).toMatchSnapshot();
  });
});
