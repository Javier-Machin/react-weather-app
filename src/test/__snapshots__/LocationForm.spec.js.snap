// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationForm should render correctly 1`] = `
<form
  className="location-form"
  onSubmit={[Function]}
>
  <input
    className="location-form__input"
    name="location"
    onChange={[Function]}
    placeholder="enter a city"
    type="text"
  />
  <input
    className="location-form__btn"
    type="submit"
    value="Check weather â†’"
  />
</form>
`;
